crates/lib/src/stdlib/tags/include_tag.rs

```diff
@@ -23,11 +25,31 @@ impl Renderable for Include {
                 .into_err();
         }
         let name = value.to_kstr().into_owned();
+
         runtime.run_in_named_scope(name.clone(), |mut scope| -> Result<()> {
+            // if there our additional varaibles creates a include object to access all the varaibles
+            // from e.g. { include 'image.html' path="foo.png" }
+            // then in image.html you could have <img src="{{include.path}}" />
+            if !self.vars.is_empty() {
+                let mut helper_vars = Object::new();
+
+                for (id, val) in &self.vars {
+                    helper_vars.insert(
+                        id.to_owned().into(),
+                        val.try_evaluate(scope)
+                            .ok_or_else(|| Error::with_msg("failed to evaluate value"))?
+                            .into_owned(),
+                    );
+                }
+
+                scope.stack_mut().set("include", Value::Object(helper_vars));
```

Comment:
Based on the test changes and looking at the Ruby implementation, it looks like we need to set the values directly on  the `stack_mut` rather than as a child of `include`
https://github.com/Shopify/liquid/blob/master/lib/liquid/tags/include.rb#L75